src/server/game/Movement/MovementGenerators/WaypointMovementGenerator.cpp

```diff
@@ -404,6 +404,9 @@ void FlightPathMovementGenerator::DoFinalize(Player* player)
     }
 
     player->RemoveFlag(PLAYER_FLAGS, PLAYER_FLAGS_TAXI_BENCHMARK);
+
+    if (ShapeshiftForm form = player->GetShapeshiftForm())
```

Comment:
Replace this with `player->RestoreDisplayId()`, will take care of transform and shapeshifts